python/lsst/pipe/tasks/multiBand.py

```diff
@@ -973,16 +973,18 @@ def run(self, patchRefList, psfCache=100):
         """
         if self.config.simultaneous:
             # Use SCARLET to simultaneously deblend across filters
-            msg = "Simultaneous deblending will be implemented once DM-15106 is completed"
-            raise NotImplementedError(msg)
-
-            # implement deblender here
-            # ...
-
-            # this is the proper code to write
-            fluxCatalogs, templateCatalogs = self.multibandBeblend.run(exposures, sources)
-            for n in len(patchRefList):
-                self.write(patchRefList[n], fluxCatalogs[n], templateCatalogs[n])
+            filters = []
+            exposures = []
+            for patchRef in patchRefList:
+                exposure = patchRef.get(self.config.coaddName + "Coadd_calexp", immediate=True)
+                filters.append(patchRef.dataId["filter"])
+                exposures.append(exposure)
+            # The input sources are the same for all bands, since it is a merged catalog
```

Comment:
Are you sure this will always be a merged catalog? what happens when it is not one

Reply:
This is the same way `MeasureMergedCoaddSourcesTask` was executed, so the input is always a merged catalog (see https://github.com/lsst/pipe_tasks/blob/tickets/DM-15104/python/lsst/pipe/tasks/multiBand.py#L1019).